orb-connd/src/service/netconfig.rs

```diff
@@ -0,0 +1,205 @@
+use super::wifi;
+use crate::network_manager::WifiSec;
+use base64::Engine as _;
+use chrono::{DateTime, Utc};
+use color_eyre::{
+    eyre::{bail, eyre, Context, ContextCompat},
+    Result,
+};
+use orb_info::orb_os_release::OrbRelease;
+use p256::{
+    ecdsa::{signature::Verifier, Signature, VerifyingKey},
+    pkcs8::DecodePublicKey,
+};
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct NetConfig {
+    pub wifi_credentials: Option<wifi::Credentials>,
+    pub wifi_enabled: Option<bool>,
+    pub smart_switching: Option<bool>,
+    pub airplane_mode: Option<bool>,
+    pub created_at: DateTime<Utc>,
+}
+
+impl NetConfig {
+    pub fn parse(netconfig_str: &str) -> Result<NetConfig> {
+        let netconfig_str = netconfig_str.replace("WIFI:", "");
+        let map: HashMap<_, _> = netconfig_str
+            .split(";")
+            .flat_map(|entry| entry.split_once(":"))
+            .collect();
```

Comment:
If there is an entry without a colon, we ignore it. Perhaps this is fine.
But what if there is a password that contains a colon? This is quite common.

Reply:
Ok, a password with a colon would pass, because we use `split_once` here, and it would be something like this: `P:abc:def`

Reply:
But what if the password contains `;`? What if it contains `\"`? What if it's quoted?